src/mongoose_listener.erl

```diff
@@ -50,3 +56,18 @@ stop_listener(Opts) ->
     ListenerId = mongoose_listener_config:listener_id(Opts),
     supervisor:terminate_child(mongoose_listener_sup, ListenerId),
     supervisor:delete_child(mongoose_listener_sup, ListenerId).
+
+%% Return deduplicated instrumentation specs.
+%% Each listener module could be started more than on different ports.
+-spec instrumentation([options()]) -> [mongoose_instrument:spec()].
+instrumentation(Listeners) ->
+    lists:usort([Spec || Listener <- Listeners, Spec <- listener_instrumentation(Listener)]).
+
+-spec listener_instrumentation(options()) -> [mongoose_instrument:spec()].
+listener_instrumentation(Opts = #{module := Module}) ->
+    case erlang:function_exported(Module, instrumentation, 1) of
```

Comment:
This wouldn't work if module is not loaded. I know it shouldn't be the case, but in general we are using `mongoose_lib:is_exported`.